### DISCUSSION TITLE
My theme no longer working after moving to zim

### DISCUSSION THREAD

**Discussion by RayPS** [zimfw/zimfw#485] [Q&A] [answered]
I'm previously a zgen user, I'm moving to zim today, everything works perfectly except my theme is no longer working, I assume zim is compatible with any zsh-theme right?
The link to my theme: https://gist.github.com/RayPS/49b2400e3d5fd74273d00925141d4b04
I expect it should looks like the screen shot in the first comment of that gist.
But it looks like this right now with zim:
![Screenshot 2022-11-06 at 22 19 19@2x](https://user-images.githubusercontent.com/1941540/200176452-dfa7654a-a848-4006-825a-033b5cb8bf49.png)
I appreciate any help!

**Comment by ericbn** [comment, marked as answer]
Hi @RayPS. You'll need to change a few things in your theme code:

* Add `setopt nopromptbang prompt{cr,percent,sp,subst}` so prompt is expanded correctly.
* Replace `%{$fg[X]%}` by `%F{X}` and `%{$reset_color%}` by `%f`. Or add `autoload -Uz colors && colors`. I prefer the former.
* Replace usages of [ohmyzsh's git lib](https://github.com/ohmyzsh/ohmyzsh/blob/master/lib/git.zsh) by [Zim's git-info](https://github.com/zimfw/git-info). Check the [git-info module documentation](https://github.com/zimfw/git-info/blob/master/README.md) and the source code of [some of our themes](https://zimfw.sh/docs/themes/) to check how it works.
